source/lib/course-search/load-cached-courses.js

```diff
@@ -0,0 +1,26 @@
+// @flow
+
+import type {CourseType, TermType} from './types'
+import {loadTermsFromStorage} from './update-course-storage'
+import {COURSE_STORAGE_DIR} from './urls'
+import RNFS from 'react-native-fs'
+
+export async function loadCachedCourses(): Array<CourseType> {
+	const terms: Array<TermType> = await loadTermsFromStorage()
+	let courses: Array<CourseType> = []
+
+	for (let i = 0; i < terms.length; i++) {
+		let newCourses: Array<CourseType> = await loadTermCoursesFromStorage(
+			terms[i],
+		)
+		courses.push(...newCourses)
+	}
+	// console.log(courses)
+	return courses
+}
+
+async function loadTermCoursesFromStorage(term: TermType): Array<CourseType> {
+	return RNFS.readFile(COURSE_STORAGE_DIR + term.path).then(async (contents) => {
```

Comment:
RNFS.readFile() always returns a Promise, which is why I made these functions async..I'm not too familiar with how to properly deal with promises..is there a better way than using async/await to unwrap the Promises? @hawkrives

Reply:
I think I have to use await when I'm looping through the terms, otherwise the Promises won't be resolved before the loop exits and it'll return undefined..right?

Reply:
You only need to mark a function `async` when you use the `await` keyword. An `async` function is a function that implicitly returns a promise, but if you're dealing with promises from other sources, or using `.then` and whatnot, you don't need to `async` the surrounding function.